javascript/code-challenges/challenges-02.test.js

```diff
@@ -155,11 +155,9 @@ const snorlaxAbilities = {
 const extractAbilities = (arr) => {
   return arr.map(value => {
     let temp = '';
-    for (let a in value) {
-      for (let b of Object.values(value)) {
-        if (typeof b === 'object')
-          temp = b.name;
-      }
+    for (let b of Object.values(value)) {
```

Comment:
So you're already looping with your map, there's no reason to loop again. You can set the properties of your new object as you map through the array and then return that array of objects, which will make your solution exponentially more efficient.